### ISSUE TITLE
python bases in packages

### ISSUE THREAD

**Issue by Eldinnie** [sphinx-doc/sphinx#4944]
### Problem
When documented our library we would like the base classes to be displayed as you can import them (and they're documented aswell)

let's assume the following

```python
# telegram/base.py

class TelegramObject(object):
  pass
```

```python
#telegram/chat.py

from telegram import TelegramObject
class Chat(TelegramObject):
  pass
```

```python
#telegram/__init__.py

from .base import TelegramObject
from .chat import Chat
__all__ = ['TelegramObject', 'Chat']
```

```yaml
index.rst

.. toctree::

  telegram.telegramobject
  telegram.chat
```

```yaml
telegram.telegramobject.rst

telegram.TelegramObject
=======================

.. autoclass:: telegram.TelegramObject
    :members:
    :show-inheritance:
```

```yaml
telegram.chat.rst

telegram.Chat
=============

.. autoclass:: telegram.Chat
    :members:
    :show-inheritance:
```
#### Procedure to reproduce the problem
run sphinx-build -b html

#### Error logs / results
Chat shows it's base as:
`telegram.base.TelegramObject`
http://python-telegram-bot.readthedocs.io/en/stable/telegram.chat.html

#### Expected results
I would like Chat to show it's base as `telegram.TelegramObject`
That way it would link through and we can properly show inheritence
There's also a issue #3014 which would also be fine so we can fix it ourselves

### Reproducible project / your project
https://github.com/python-telegram-bot/python-telegram-bot/tree/master/docs

### Environment info
- OS: Linux/Win/whatever runs on rtd
- Python version: 2.7, 3.6 or 3.7
- Sphinx version: 1.6.5, 1.7.2

**Comment by MuellerSeb** [comment]
https://github.com/sphinx-doc/sphinx/issues/5589 is similar.

**Comment by tk0miya** [comment]
Now I merged #9026 that generates :canonical: option for the imported classes. It allows to refer to the python objects by its canonical name. It will resolve this problem.
Thanks,

**Comment by tk0miya** [comment]
Ah, sorry. #9026 does not resolve this issue. It allows to make a hyperlink as `telegram.base.TelegramObject`, not `telegram.TelegramObject`. So this is not resolved yet. So reopening now.
